reconcile/utils/slack_api.py

```diff
@@ -80,16 +80,29 @@ def get_usergroup(self, handle):
         [usergroup] = usergroup
         return usergroup
 
-    @retry()
+    @retry(exceptions=SlackAPIRateLimitedException, max_attempts=5)
     def update_usergroup(self, id, channels_list, description):
         channels = ','.join(channels_list)
-        self.sc.api_call(
+        response = self.sc.api_call(
             "usergroups.update",
             usergroup=id,
             channels=channels,
             description=description,
         )
 
+        error = response.get('error')
+
+        if error == 'ratelimited':
+            retry_after = response['headers']['retry-after']
+            time.sleep(int(retry_after))
```

Comment:
`exceptions` in `retry()` controls which exceptions we will retry for, rather than which will not be retried.  The latter is controlled by `no_retry_exceptions`.  We are sleeping twice because `retry()` sleeps and I'm sleeping for the amount of time that the Slack API tells us to.  I'm not super worried about this at the moment because `retry()` does a linear sleep increase, so it shouldn't be too long.  Additionally, I'm hoping to revamp this whole retry logic in the next sprint by upgrading to the latest Slack SDK (supports retries automatically).

Reply:
Also, the `test_slack_api_update_usergroup_rate_limit_raise` and related tests should prove that the retry is actually happening because it checks how many times the SlackApi client is called.  In addition, this logic has all pretty much been lifted from `update_usergroup_users` which is currently working without issue.  I'd definitely generalize this all if I wasn't intending to replace it in the next few weeks.

Reply:
Thanks for the explanation. I totally overlooked `exceptions` vs `no_retry_exceptions` from the retry decorator